lightning/src/ln/channelmanager.rs

```diff
@@ -10536,10 +10536,7 @@ where
 					},
 					None => match invoice_request.verify(expanded_key, secp_ctx) {
 						Ok(invoice_request) => invoice_request,
-						Err(()) => {
-							let error = Bolt12SemanticError::InvalidMetadata;
-							return responder.respond(OffersMessage::InvoiceError(error.into()));
-						},
+						Err(()) => return ResponseInstruction::NoResponse,
```

Comment:
Currently, this will only be triggered for older offers. When we add support for producing offers without blinded paths (i.e. with a public node id), it will also be hit. So at that point someone could de-anonymize the older offer by sending an invoice request for it to the node id given in the newer offer.

Reply:
Yeah, probably not useful to send an `InvoiceError`. The only time those should ever fail is if we are given an invoice request for an invalid offer or an invalid path id used. Older offers should still work.